### DISCUSSION TITLE
[Dropdown Menu] How do I prevent click to open if user clicked and dragged?

### DISCUSSION THREAD

**Discussion by matheusbaumgart** [radix-ui/primitives#2332] [Help]
The button (trigger) that opens the menu is also used to click and drag in my application. How do I prevent the menu to open if the user starts a drag action?

Cheers team 🙏

**Comment by hyjoong** [comment]
Did you wrap the button with DropdownMenuTrigger? If you structure the dropdown as shown in the link below, it should behave as shadn ui dropdown : https://ui.shadcn.com/docs/components/dropdown-menu

```javascript
import * as DropdownMenuPrimitive from "@radix-ui/react-dropdown-menu";

const DropdownMenuTrigger = DropdownMenuPrimitive.Trigger;

 <DropdownMenuTrigger>
          <Button>
            open dropdown
          </Button>
        </DropdownMenuTrigger>
        <DropdownMenuContent>
....
```

**Reply by matheusbaumgart** [reply]
Hi @hyjoong that dropdown from shadcn and the code above have the same issue, if the user **clicks then drags** the dropdown opens. I want to prevent it from opening if the user is dragging.

**Reply by matheusbaumgart** [reply]
Any ideas here @hyjoong?
